### PR TITLE
DEV-3175-new-go-exercise-printfirsthalf

### PR DESCRIPTION
**Pull request by zainabdnaya**

### CODE REVIEW

File: solutions/printfirsthalf.go [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package solutions
+
+import "strings"
+
+func Printfirsthalf(str string) string {
```
Comment (jotapero) [inline solutions/printfirsthalf.go:5]: change to "PrintFirstHalf" - maintain PascalCase formatting

File: tests/printfirsthalf_test/main.go [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+	student "student"
+
+	"github.com/01-edu/go-tests/lib/challenge"
+	"github.com/01-edu/go-tests/solutions"
+)
+
+func main() {
+	table := []string{"hello", "", "Kimetsu no Yaiba", "Z", "123@live.fr", "write %d ==> 45m$", "-552",}
+	for _, s := range table {
+		challenge.Function("Printfirsthalf", student.Printfirsthalf, solutions.Printfirsthalf, s)
```
Comment (jotapero) [inline tests/printfirsthalf_test/main.go:13]: Same here with case formatting "PrintFirstHalf

**Review by jotapero — CHANGES_REQUESTED** [review]

File: solutions/printfirsthalf.go (hunk 1 above)
Comment (zainabdnaya) [reply solutions/printfirsthalf.go]: it's fixed

File: tests/printfirsthalf_test/main.go (hunk 2 above)
Comment (zainabdnaya) [reply tests/printfirsthalf_test/main.go]: it's fixed

**Review by jotapero — APPROVED** [review]

**Pull request closed without merging**